crates/agentgateway/proto/resource.proto

```diff
@@ -7,6 +7,8 @@ import "google/protobuf/wrappers.proto";
 import "google/protobuf/any.proto";
```

Comment:
Can we delete the old proto files (https://github.com/agentgateway/agentgateway/blob/4e27972c78ee980c1e6fd599cc47786850db2add/crates/agentgateway/proto/listener.proto), it looks like they're causing conflicts with the new Listener in resource.proto